src/Compatibility/Microsoft.DotNet.PackageValidation/ValidatePackage.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.ApiCompatibility;
+using Microsoft.NET.Build.Tasks;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    public class ValidatePackage : TaskBase
+    {
+        [Required]
+        public string PackagePath { get; set; }
+
+        public string RuntimeGraph { get; set; }
+
+        public string NoWarn { get; set; }
+
+        public bool RunApiCompat { get; set; }
+
+        public bool ValidateWithPreviousVersion { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            Package package = NupkgParser.CreatePackage(PackagePath, RuntimeGraph);
+
+            LogErrors(new CompatibleTfmValidator(NoWarn, null, RunApiCompat).Validate(package));
+            LogErrors(new PreviousVersionValidator(package, NoWarn, null, RunApiCompat).Validate(package));
+
+            foreach (var apiCompatError in new CompatibleFrameworkInPackageValidator(NoWarn, null).Validate(package))
+            {
+                Log.LogError(apiCompatError.CompatibilityReason);
+                Log.LogError(apiCompatError.Header);
+                foreach (var error in apiCompatError.Differences.Differences)
+                {
+                    Log.LogError(error.ToString());
+                }
```

Comment:
It seems odd to me that we're capturing these in a collection in memory rather than logging them as we see them (by passing down a logger, or something).  I might be missing the reason for that.

Reply:
I checked and u r right we can change them to log as we go, still need to figure how to test the failures.